### ISSUE TITLE
Commit hyperlink not working 🐛 

### ISSUE THREAD

**Issue by porridgewithraisins** [dandavison/delta#1628]
I have 

```
[delta]
  hyperlinks = true
```

in my `~/.gitconfig`

I have ~/e2ee.js/.git/config
```
[remote "origin"]
        url = https://github.com/porridgewithraisins/e2ee.js
        fetch = +refs/heads/*:refs/remotes/origin/*
```

Files are hyperlinked to vscode as I have set `hyperlinks-file-link-format` to the `vscode://` handler.

However, the commit hash isn't rendering as a hyperlink.

<img width="1438" alt="image" src="https://github.com/dandavison/delta/assets/72668511/53516224-c9ca-479d-8e88-0b0486bd4a2b">

I am using iterm, zsh here.

**Comment by gczajkowskiTT** [comment]
Same for me

**Comment by dandavison** [comment]
Hi @porridgewithraisins, @gczajkowskiTT let's look at the raw bytes that delta's emitting. Can you post the output of

```
git show | delta | cat -A
```

(or `bat -A` if you have bat installed).

**Comment by gczajkowskiTT** [comment]
In Linux under Kitty (which is closest to functioning)
```
<E2><90><8A>
<E2><90><8A>
<E2><90>[34m\u{2500}\u{2500}\u{2500}\u{2500}\u{2500}\u{2500}\u{2500}\u{2500}\u{2500}\u{2500}\u{2500}\u{2500}\u{2500}\u{2500}\u{2500}\u{2500}<E2><90>[0m<E2><90>[34m\u{2510[34m[0m[34m}<E2><90>[0m<E2><90><8A>
<E2><90>[31m\u{2022}<E2><90>[0m<C2><B7><E2><90>8;;file:///proj_soc/user_dev/gczajkowski/ttdelivery/.gitignore<E2><90><E2><90>[31m.gitignore<E2><90>[0m:<E2><90>[38;2;6;122;0m1<E2><90>[0m<E2>
[31m[0m[31m[0m[38;2;6;122;0m[0m<90>8;;<E2><90>:<C2><B7><E2><90>[34m\u{2502}<E2><90>[0m<E2><90><8A>
<E2><90>[34m\u{2500}\u{2500}\u{2500}\u{2500}\u{2500}\u{2500}\u{2500}\u{2500}\u{2500}\u{2500}\u{2500}\u{2500}\u{2500}\u{2500}\u{2500}\u{2500}<E2><90>[0m<E2><90>[34m\u{2518[34m[0m[34m}<E2><90>[0m<E2><90><8A>
<E2><90>[38;2;117;113;94m#<C2><B7>Linux<E2><90>[0m<E2><90><8A>
```

**Comment by porridgewithraisins** [comment]
I now use a linux laptop and it works for me. However, in the few months I used a mac, it did not work at all. Not able to test now unfortunately.

**Comment by dandavison** [comment]
Has anyone observed that the commit hyperlink doesn't work while other hyperlinks _do_ work? Or are you saying that all hyperlinks (commit, files, line numbers) don't work when there's a problem?

Also if the latter, are you using tmux?

**Comment by gczajkowskiTT** [comment]
No hyperlinks are working for me and this is without tmux. Part of the problem looked to be git `delta.features = decorations` .. as it effected the 'shell output code' .. so disabled it, however it still mostly garbage

**Comment by dandavison** [comment]
> No hyperlinks are working for me and this is without tmux

If you do
```
printf '\e]8;;vscode://file//Users/jaminthornsberry/.gitconfig:10\e\\Take me to git config!\e]8;;\e\n'
```

then that hyperlink works, right?

If so, then if you capture the bytes emitted by delta (`git show | delta | hexyl or cat -A etc`) and compare to that `printf`, do you see a difference?

**Comment by gczajkowskiTT** [comment]
> ```printf '\e]8;;vscode://file//Users/jaminthornsberry/.gitconfig:10\e\\Take me to git config!\e]8;;\e\n'```
> then that hyperlink works, right?

Yes, specifically in VSCode terminal in MacOS. But I can't use these for Linux links. 

> Hi @porridgewithraisins, @gczajkowskiTT let's look at the raw bytes that delta's emitting. Can you post the output of
> 
> ```
> git show | delta | cat -A

No such option in MacOS `cat`
$ git show | delta | cat -A                                                                                
cat: illegal option -- A

> ```
> 
> (or `bat -A` if you have bat installed).

With `bat -A` under VSCode in MacOS there is no link
<img width="1140" alt="image" src="https://github.com/user-attachments/assets/4cecf1bf-1f2a-4c1f-b2b6-5e2c12f1f290">

**Comment by dandavison** [comment]
> With `bat -A` under VSCode in MacOS there is no link

Hmm. I haven't yet come up with a hypothesis for why that would be. It's not the case in my VSCOde terminal on MacOS:

<img width="1374" alt="image" src="https://github.com/user-attachments/assets/7373c086-4b13-4303-b3c4-24eeff42e17b">

Here's the code. Can you think what might be going on? It's definitely finding your git config is it? The code shows that the hyperlink will be added if the hyperlink format setting is in effect or if it can find git config and find a git remote for the repo. Are you talking about a situation where the commit hyperlink works fine in a standard terminal emulator on the same machine and then suddenly stops working in VSCode?

https://github.com/dandavison/delta/blob/1fb6f99ec498150f552c1c8b5440152c0c2aade7/src/features/hyperlinks.rs#L22-L43

**Comment by dandavison** [comment]
I'm guessing that if you set `hyperlinks-commit-link-format` it works in VSCode, right? It's something to do with the git remote not being identified in the vscode context?

**Comment by gczajkowskiTT** [comment]
Here is something interesting `hyperlinks = true` alone does not enable links by default. However, if I add a hard-coded repo specific `hyperlinks-commit-link-format = "https://<gitlab.server.instance/path/to/project/-/commit/{commit}/"`

Then 
1) delta attempts to add the links to `git blame` 
2) the links have multiple shaids? 
https://<gitlab.server.instance/path/to/project/-/commit/{commit}/someothershaid
3) the links are still very broken in linux kitty

From the documentation:
> Commit hashes link to GitHub/GitLab/Bitbucket (use hyperlinks-commit-link-format for full control).

So this won't work for our private gitlab instances

>It's something to do with the git remote not being identified in the vscode context?

Not sure. If I add hyperlinks-commit-link-format to both Linux and MacOS, the links are added.However I have no clue what to add for hyperlinks-commit-link-format in a local repo.

On MacOS: `git config --local --get remote.origin.url` returns {host}:/path/to/clone
On Linux: `git config --local --get remote.origin.url` correctly returns the *URI* which is not a URL since it uses `git@internal.gitlab.server.instance:<path/project.git>` scheme. And we have hundreds of projects and several gitlab private instances so I would probably have to figure out a repo URI/URL dynamically per OS

(For my reference I was looking at https://github.com/dandavison/delta/issues/613)
